### DISCUSSION TITLE
Prisma not loading precompiled binaries

### DISCUSSION THREAD

**Discussion by Chaphasilor** [prisma/prisma#3858] [Q&A]
Hey, I have an issue with prisma :)

I'm trying to set up https://github.com/mikecao/umami on a Raspberry Pi 4 (arm64, running in 32-bit mode).
The problem is, there are no precompiled binaries for arm (see https://github.com/prisma/prisma/issues/861 ).

So I've now compiled them myself and have them available on my machine. But when I run `npm run build` on umami, prisma still tries to download prebuilts, and fails with a fatal error because of the 404.
This shouldn't be happening, because I have the environment variables set to point to my binaries (see https://github.com/prisma/prisma/issues/1329 ).

 Here's the output when installing:
```bash
Database type detected: postgresql
Copied /home/code/umami/prisma/schema.postgresql.prisma to /home/code/umami/prisma/schema.prisma
> umami@0.74.0 build-db-client /home/code/umami
> dotenv prisma generate
Environment variables loaded from prisma/.env
Warning Precompiled binaries are not available for arm.
> Downloading Prisma engines for arm-openssl-1.1.x [                    ] 0%Error: Not Found https://binaries.prisma.sh/all_commits/e6c9b4b2b7fa162d0d459d1863321f547498fcfe/arm-openssl-1.1.x/prisma-fmt.gz
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! umami@0.74.0 build-db-client: `dotenv prisma generate`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the umami@0.74.0 build-db-client script.
```

And here are the contents of prisma/.env :
```bash
PRISMA_QUERY_ENGINE_BINARY=/home/code/prisma-engines/target/release/query-engine
PRISMA_INTROSPECTION_ENGINE_BINARY=/home/code/prisma-engines/target/release/introspection-engine
PRISMA_MIGRATION_ENGINE_BINARY=/home/code/prisma-engines/target/release/migration-engine
```

You can see that the envvars are loaded:
> `Environment variables loaded from prisma/.env`

Any ideas how to fix this?

**Comment by ryands17** [comment]
Hey @Chaphasilor 👋 
Could you check if the solution is in any of these issues?
https://github.com/prisma/prisma/issues?q=is%3Aissue+raspberry

**Comment by Chaphasilor** [comment]
Hey, of course I check these issues before :)

I've even referenced some of them in my issue above!

They helped me get to this stage, but now I'm seeing different behavior from what is described in these issues... 

Edit: seems like I've missed a detail, I have a few new ideas I'm trying...

**Comment by Chaphasilor** [comment]
The problem I'm facing is that the method described [here](https://github.com/prisma/prisma/issues/1505#issuecomment-634154657) is for an older version of Prisma, and since then some things seem to have changed, including new contribution docs (I used the old version as a reference when trying the solution), a few changed directory names, etc.  

However, the project I am trying to set up (https://github.com/mikecao/umami/) is using a newer version of Prisma, which I think is not compatible with the old solution.  
I believe the problem is how umami uses Prisma as a dependency, causing pnpm to not be able to effectively link my custom build. This is only a theory though, I honestly don't really know what I'm doing xD

**Reply by ryands17** [reply]
Could you try [this](https://gitlab.com/npappas/prisma-engines) from one of our community members and check if this works for you?

**Reply by Chaphasilor** [reply]
I already compiled the binaries :)
I just can't get Prisma to use them :/

Or is there a guide for that in the Gitlab that I didn't see?

**Reply by ryands17** [reply]
[This](https://www.prisma.io/docs/understand-prisma/under-the-hood#using-custom-engine-binaries) is the way to make Prisma use them, but I'm assuming you've already done that.

**Reply by Chaphasilor** [reply]
Yep. In my output above you can see that dotenv has successfully loaded the variables before calling Prisma. I've also confirmed the path to the binaries is correct.  

I believe the problem is in the command `dotenv prisma generate`, in that it's not calling my custom prisma cli. I tried linking it with pnpm, but no luck...

**Reply by ryands17** [reply]
Shouldn't it be `dotenv -- prisma generate`?
Also in your very first comment, the env vars are loaded from `prisma/.env`which is the default location. Do you have your env file anywhere else (like in the root of your project)?

**Reply by Chaphasilor** [reply]
I tried it with `dotenv --`, but that didn't change anything.  
I've also sprinkled the .env file all around the projects ^^. Project root, custom prisma build root, etc.

I've now tried specifying my custom build manually, so `dotenv -- /home/code/prisma_custom/src/packages/cli/build/index.js generate`.  
However, I'm still getting the `Unknown binaryTargets arm-openssl-1.1.x`-error. Which makes me think that maybe the custom build didn't work properly?  
I used https://github.com/prisma/prisma/blob/df15ba03133cb67f1d7c94564da6a6b54bbd2de5/CONTRIBUTING.md, as this was the latest version of the file when @fabianmoronzirfas forked the repo. I'm not completely sure which 'first steps' he performed, and if I have to perform them twice as well (given that the source code is already modified)
